code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushPayload.java

```diff
@@ -240,6 +268,19 @@ private int getNotificationImportanceFromString(final String priority) {
         return resolvedImportance;
     }
 
+    // Returns the notification visibility from the string
+    // If the string is null or not a valid visibility, returns Notification.VISIBILITY_PRIVATE
+    //
+    // @param visibility string representing the visibility of the notification
+    // @return int representing the visibility of the notification
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private int getNotificationVisibilityFromString(final String visibility) {
+        if (visibility == null) return Notification.VISIBILITY_PRIVATE;
```

Comment:
check for empty string as well: StringUtils.isNullOrEmpty as an empty string will not match a key in the notificationVisibilityMap